mithril-core/src/error.rs

```diff
@@ -126,6 +130,7 @@ impl<D: Digest> From<RegisterError> for StmSignatureError<D> {
             RegisterError::KeyInvalid(e) => Self::IvkInvalid(e.vk),
             RegisterError::KeyRegistered(_) => unreachable!(),
             RegisterError::UnregisteredInitializer => unreachable!(),
+            RegisterError::KeyNonExisting => todo!(),
```

Comment:
This seems also unreachable, right?
```suggestion
            RegisterError::KeyNonExisting => unreachable!(),
```

Reply:
This error should not exist in the `mithril-core` library IMHO. 
I moved it back to `mithril-common` where the existence in the stake distribution is done :ok_hand: